src/render.rs

```diff
@@ -197,95 +240,35 @@ impl<'a> Renderer<'a> {
-                                            _ => unreachable!()
-                                        };
-                                        return Ok(result);
-                                    },
-                                    _ => unreachable!()
-                                }
-                            },
-                            Int(n) => {
-                                // rhs MUST be a number
-                                let l = n as f32; // TODO: that's going to cause issues
-                                let r = try!(self.eval_math(&rhs));
-                                let result = match operator.as_str() {
-                                    "==" => (l - r).abs() < EPSILON,
-                                    "!=" => (l - r).abs() > EPSILON,
-                                    _ => unreachable!()
-                                };
-                                return Ok(result);
-                            },
-                            Float(l) => {
-                                // rhs MUST be a number
-                                let r = try!(self.eval_math(&rhs));
-                                let result = match operator.as_str() {
-                                    "==" => (l - r).abs() < EPSILON,
-                                    "!=" => (l - r).abs() > EPSILON,
-                                    _ => unreachable!()
-                                };
-                                return Ok(result);
-                            },
-                            Math { .. } => {
-                                // rhs MUST be a number
-                                let l = try!(self.eval_math(&lhs));
-                                let r = try!(self.eval_math(&rhs));
-                                let result = match operator.as_str() {
-                                    "==" => (l - r).abs() < EPSILON,
-                                    "!=" => (l - r).abs() > EPSILON,
-                                    _ => unreachable!()
-                                };
-                                return Ok(result);
-                            },
-                            _ => unreachable!()
+                        let mut lhs_val = try!(self.eval_expression(*lhs));
+                        let mut rhs_val = try!(self.eval_expression(*rhs));
+
+                        // Monomorphize number vals.
+                        if lhs_val.is_number() || rhs_val.is_number() {
+                            if !lhs_val.is_number() || !rhs_val.is_number() {
+                                return Ok(false);
+                            }
+
+                            // Since Tera only support 32 bit integers, this
+                            // actually preserves all of the precision. If Tera
+                            // switches to 64-bit values, use std::f32::EPSILON
+                            // to get an approximation as before.
```

Comment:
that's something I was thinking about actually, I'd be tempted to always use 64-bit